# HG changeset patch
# User jwe
# Date 784831025 0
#      Mon Nov 14 16:37:05 1994 +0000
# Node ID 1677bb6533fb2c8f726abad2deafd30eb926a447
# Parent  ba98491f97b20fec15a3254e8d3ad1678abf24bc
[project @ 1994-11-14 16:37:05 by jwe]
Initial revision

diff --git a/scripts/strings/strcat.m b/scripts/strings/strcat.m
new file mode 100644
--- /dev/null
+++ b/scripts/strings/strcat.m
@@ -0,0 +1,42 @@
+# Copyright (C) 1994 John W. Eaton
+# 
+# This file is part of Octave.
+# 
+# Octave is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2, or (at your option) any
+# later version.
+# 
+# Octave is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with Octave; see the file COPYING.  If not, write to the Free
+# Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+
+function st = strcat (s, t, ...)
+
+  if (nargin > 1)
+    if (isstr (s) && isstr (t))
+      tmpst = [s, t];
+    else
+      error ("strcat: all arguments must be strings");
+    endif
+    n = nargin - 2;
+    while (n--)
+      tmp = va_arg ();
+      if (isstr (tmp))
+        tmpst = [tmpst, tmp];
+      else
+        error ("strcat: all arguments must be strings");
+      endif
+    endwhile
+  else
+    usage ("strcat (s, t, ...)");
+  endif
+
+  st = tmpst;
+
+endfunction
